Dumper::formatValueForDump() Argument #1 ($value) must be of type ?string, int given

### Affected version(s)

4.13.x-dev

### Description

When using `contao:migrate (or `backup:create`) the following error happens in Contao `4.13.x-dev`:

```
[ERROR] Contao\CoreBundle\Doctrine\Backup\Dumper::formatValueForDump():      
         Argument #1 ($value) must be of type ?string, int given, called in   
         vendor\contao\contao\core-bundle\src\Doctrine\Backup\Dumper.php on line 122 
```

The column in question is `tl_article.id` and the value to be formated is an integer of value `1`. This error was introduced in https://github.com/contao/contao/commit/52753e05c0efdcb6a6cb28e2c75665d544ece208

/cc @Toflar

--- a/core-bundle/src/Doctrine/Backup/Dumper.php
+++ b/core-bundle/src/Doctrine/Backup/Dumper.php
@@ -133,6 +133,9 @@ class Dumper implements DumperInterface
         }
     }
 
+    /**
+     * @param string|int|float|null $value
+     */
     private function formatValueForDump($value, int $columnBindingType, bool $isUtf8Charset, Connection $connection): string
     {
         if (null === $value) {

--- a/core-bundle/src/Doctrine/Backup/Dumper.php
+++ b/core-bundle/src/Doctrine/Backup/Dumper.php
@@ -133,11 +133,13 @@ class Dumper implements DumperInterface
         }
     }
 
-    private function formatValueForDump(?string $value, int $columnBindingType, bool $isUtf8Charset, Connection $connection): string
+    private function formatValueForDump($value, int $columnBindingType, bool $isUtf8Charset, Connection $connection): string
     {
         if (null === $value) {
             return 'NULL';
         }
+
+        $value = (string) $value;
 
         if ('' === $value) {
             return "''";

--- a/core-bundle/tests/Doctrine/Backup/DumperTest.php
+++ b/core-bundle/tests/Doctrine/Backup/DumperTest.php
@@ -118,6 +118,50 @@ class DumperTest extends ContaoTestCase
             ],
         ];
 
+        yield 'Table with float and integer data' => [
+            [
+                new Table('tl_page', [
+                    new Column('stringCol', Type::getType(Types::STRING)),
+                    new Column('integerCol', Type::getType(Types::INTEGER)),
+                    new Column('floatCol', Type::getType(Types::FLOAT)),
+                    new Column('bigintCol', Type::getType(Types::BIGINT)),
+                    new Column('decimalCol', Type::getType(Types::DECIMAL)),
+                    new Column('booleanCol', Type::getType(Types::BOOLEAN)),
+                ]),
+            ],
+            [],
+            [
+                'SELECT `stringCol` AS `stringCol`, `integerCol` AS `integerCol`, `floatCol` AS `floatCol`, `bigintCol` AS `bigintCol`, `decimalCol` AS `decimalCol`, `booleanCol` AS `booleanCol` FROM `tl_page`' => [
+                    [
+                        'stringCol' => 'value1',
+                        'integerCol' => '42',
+                        'floatCol' => '4.2',
+                        'bigintCol' => '92233720368547758079223372036854775807',
+                        'decimalCol' => '4.2',
+                        'booleanCol' => '1',
+                    ],
+                    [
+                        'stringCol' => 'value1',
+                        'integerCol' => 42,
+                        'floatCol' => 4.2,
+                        'bigintCol' => '92233720368547758079223372036854775807',
+                        'decimalCol' => 4.2,
+                        'booleanCol' => 1,
+                    ],
+                ],
+            ],
+            [
+                'SET FOREIGN_KEY_CHECKS = 0;',
+                '-- BEGIN STRUCTURE tl_page',
+                'DROP TABLE IF EXISTS `tl_page`;',
+                'CREATE TABLE tl_page (stringCol VARCHAR(255) NOT NULL, integerCol INT NOT NULL, floatCol DOUBLE PRECISION NOT NULL, bigintCol BIGINT NOT NULL, decimalCol NUMERIC(10, 0) NOT NULL, booleanCol TINYINT(1) NOT NULL) DEFAULT CHARACTER SET utf8 COLLATE `utf8_unicode_ci` ENGINE = InnoDB;',
+                '-- BEGIN DATA tl_page',
+                "INSERT INTO `tl_page` (`stringCol`, `integerCol`, `floatCol`, `bigintCol`, `decimalCol`, `booleanCol`) VALUES ('value1', 42, '4.2', '92233720368547758079223372036854775807', '4.2', 1);",
+                "INSERT INTO `tl_page` (`stringCol`, `integerCol`, `floatCol`, `bigintCol`, `decimalCol`, `booleanCol`) VALUES ('value1', 42, '4.2', '92233720368547758079223372036854775807', '4.2', 1);",
+                'SET FOREIGN_KEY_CHECKS = 1;',
+            ],
+        ];
+
         yield 'Multiple tables with data' => [
             [
                 new Table('tl_page', [new Column('foobar', Type::getType(Types::STRING))]),

